packages/openapi-generator/src/sdk-metadata/code-sample-util.ts

```diff
@@ -0,0 +1,167 @@
+  }
+  // If no closest api found, return the api with most operations
+  const sortedByOperationsLength = apis.sort((a, b) =>
+    a.operations.length > b.operations.length ? -1 : 1
+  );
+  const withGetAll = sortedByOperationsLength.filter(api =>
+    api.operations.find(operation =>
+      operation.operationId.toLocaleLowerCase('getall')
+    )
+  );
+  if (withGetAll) {
+    return withGetAll[0];
+  }
+  // return the one with most methods
+  return sortedByOperationsLength[0];
+}
+
+export function getMainOperation(api: OpenApiApi): OpenApiOperation {
+  if (api.operations.length === 1) {
+    return api.operations[0];
+  }
+  return (
+    getLevensteinClosestOperation(api.name, api.operations) ||
+    getGetAllOperation(api.operations) ||
+    getGetOperation(api.operations) ||
+    getAnyOperationWithoutParams(api.operations) ||
+    api.operations[0]
+  );
+}
+
+export function getLevensteinClosestOperation(
+  apiName: string,
+  operations: OpenApiOperation[]
+): OpenApiOperation | undefined {
+  let closestOperation: OpenApiOperation | undefined;
+  let minDistance = distanceThreshold;
+
+  operations.forEach(operation => {
+    const distance = getLevensteinDistance(apiName, operation.operationId);
+    if (distance <= minDistance) {
+      minDistance = distance;
+      closestOperation = operation;
+    }
+  });
+  return closestOperation;
+}
+
+export function getGetAllOperation(
+  operations: OpenApiOperation[]
+): OpenApiOperation | undefined {
+  return operations.find(
+    operation =>
+      operation.operationId.toLowerCase() === 'getall' &&
+      operation.pathParameters.length === 0
+  );
+}
+
+export function getGetOperation(
+  operations: OpenApiOperation[]
+): OpenApiOperation | undefined {
+  const getOperations = operations.filter(
+    operation => operation.method === 'get'
+  );
+
+  // check if get operation without parameters exists
+  // else return operation with min parameters
+  if (getOperations.length > 0) {
+    const getOperationsWithoutParams = getOperations.filter(
+      getOperation =>
+        getOperation.pathParameters.length === 0 &&
+        getOperation.queryParameters.length === 0
+    );
+    if (getOperationsWithoutParams.length > 0) {
+      return getOperationsWithoutParams[0];
+    }
+    return getOperations[0];
+  }
+  return undefined;
+}
+
+export function getAnyOperationWithoutParams(
+  operations: OpenApiOperation[]
+): OpenApiOperation | undefined {
+  return operations.find(
+    operation =>
+      operation.pathParameters.length === 0 &&
+      operation.queryParameters.length === 0
+  );
+}
+
+export function getOperationParamCode(operation: OpenApiOperation): string {
+  if (
+    operation.pathParameters?.length === 0 &&
+    operation.queryParameters?.filter(param => param.required).length === 0 &&
+    operation.requestBody === undefined
+  ) {
+    return '';
+  }
+
+  const paramSignature: string[] = [];
+  if (operation.pathParameters?.length > 0) {
```

Comment:
is the check necessary? If the array is empty it will not do something or do you check for undefined? if so just do if(operation.pathParameters)